From 72c52d0fdbef462e1c404a80543039afdc122338 Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Wed, 19 Jan 2011 15:40:56 +0100
Subject: [PATCH 6/6] alsa-mixer: Make sure capture source and input source use right path

Make sure that mic and line (with common names) use the specific
path instead of the analog-input one.

Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 .../alsa/mixer/paths/analog-input-dock-mic.conf    |    2 +
 .../alsa/mixer/paths/analog-input-front-mic.conf   |    2 +
 .../mixer/paths/analog-input-internal-mic.conf     |    4 ++
 src/modules/alsa/mixer/paths/analog-input-mic.conf |    2 +
 .../alsa/mixer/paths/analog-input-rear-mic.conf    |    2 +
 .../alsa/mixer/paths/analog-input.conf.common      |   45 --------------------
 6 files changed, 12 insertions(+), 45 deletions(-)

diff --git a/src/modules/alsa/mixer/paths/analog-input-dock-mic.conf b/src/modules/alsa/mixer/paths/analog-input-dock-mic.conf
index df8567e..74826a9 100644
--- a/src/modules/alsa/mixer/paths/analog-input-dock-mic.conf
+++ b/src/modules/alsa/mixer/paths/analog-input-dock-mic.conf
@@ -53,12 +53,14 @@ enumeration = select
 
 [Option Input Source:Dock Mic]
 name = analog-input-microphone-dock
+required-any = any
 
 [Element Capture Source]
 enumeration = select
 
 [Option Capture Source:Dock Mic]
 name = analog-input-microphone-dock
+required-any = any
 
 [Element Mic]
 switch = off
diff --git a/src/modules/alsa/mixer/paths/analog-input-front-mic.conf b/src/modules/alsa/mixer/paths/analog-input-front-mic.conf
index 516b487..6c58ece 100644
--- a/src/modules/alsa/mixer/paths/analog-input-front-mic.conf
+++ b/src/modules/alsa/mixer/paths/analog-input-front-mic.conf
@@ -53,12 +53,14 @@ enumeration = select
 
 [Option Input Source:Front Mic]
 name = analog-input-microphone-front
+required-any = any
 
 [Element Capture Source]
 enumeration = select
 
 [Option Capture Source:Front Mic]
 name = analog-input-microphone-front
+required-any = any
 
 [Element Mic]
 switch = off
diff --git a/src/modules/alsa/mixer/paths/analog-input-internal-mic.conf b/src/modules/alsa/mixer/paths/analog-input-internal-mic.conf
index 3e45bcf..70a1cd1 100644
--- a/src/modules/alsa/mixer/paths/analog-input-internal-mic.conf
+++ b/src/modules/alsa/mixer/paths/analog-input-internal-mic.conf
@@ -75,18 +75,22 @@ enumeration = select
 
 [Option Input Source:Internal Mic]
 name = analog-input-microphone-internal
+required-any = any
 
 [Option Input Source:Int Mic]
 name = analog-input-microphone-internal
+required-any = any
 
 [Element Capture Source]
 enumeration = select
 
 [Option Capture Source:Internal Mic]
 name = analog-input-microphone-internal
+required-any = any
 
 [Option Capture Source:Int Mic]
 name = analog-input-microphone-internal
+required-any = any
 
 [Element Mic]
 switch = off
diff --git a/src/modules/alsa/mixer/paths/analog-input-mic.conf b/src/modules/alsa/mixer/paths/analog-input-mic.conf
index 8565ebe..d88028b 100644
--- a/src/modules/alsa/mixer/paths/analog-input-mic.conf
+++ b/src/modules/alsa/mixer/paths/analog-input-mic.conf
@@ -53,12 +53,14 @@ enumeration = select
 
 [Option Input Source:Mic]
 name = analog-input-microphone
+required-any = any
 
 [Element Capture Source]
 enumeration = select
 
 [Option Capture Source:Mic]
 name = analog-input-microphone
+required-any = any
 
 ;;; Some AC'97s have "Mic Select" and "Mic Boost (+20dB)"
 
diff --git a/src/modules/alsa/mixer/paths/analog-input-rear-mic.conf b/src/modules/alsa/mixer/paths/analog-input-rear-mic.conf
index 45f7730..75ed61b 100644
--- a/src/modules/alsa/mixer/paths/analog-input-rear-mic.conf
+++ b/src/modules/alsa/mixer/paths/analog-input-rear-mic.conf
@@ -53,12 +53,14 @@ enumeration = select
 
 [Option Input Source:Rear Mic]
 name = analog-input-microphone-rear
+required-any = any
 
 [Element Capture Source]
 enumeration = select
 
 [Option Capture Source:Rear Mic]
 name = analog-input-microphone-rear
+required-any = any
 
 [Element Mic]
 switch = off
diff --git a/src/modules/alsa/mixer/paths/analog-input.conf.common b/src/modules/alsa/mixer/paths/analog-input.conf.common
index 0b2cfd9..9416577 100644
--- a/src/modules/alsa/mixer/paths/analog-input.conf.common
+++ b/src/modules/alsa/mixer/paths/analog-input.conf.common
@@ -66,42 +66,18 @@ enumeration = select
 name = input-microphone
 priority = 20
 
-[Option Input Source:Mic]
-name = input-microphone
-priority = 20
-
 [Option Input Source:Microphone]
 name = input-microphone
 priority = 20
 
-[Option Input Source:Front Mic]
-name = input-microphone
-priority = 19
-
 [Option Input Source:Front Microphone]
 name = input-microphone
 priority = 19
 
-[Option Input Source:Int Mic]
-name = input-microphone
-priority = 19
-
-[Option Input Source:Internal Mic]
-name = input-microphone
-priority = 19
-
-[Option Input Source:Rear Mic]
-name = input-microphone
-priority = 19
-
 [Option Input Source:Internal Mic 1]
 name = input-microphone
 priority = 19
 
-[Option Input Source:Line]
-name = input-linein
-priority = 18
-
 [Option Input Source:Line-In]
 name = input-linein
 priority = 18
@@ -135,21 +111,12 @@ name = input
 [Option Capture Source:Line/Mic]
 name = input
 
-[Option Capture Source:Mic]
-name = input-microphone
-
 [Option Capture Source:Microphone]
 name = input-microphone
 
-[Option Capture Source:Int Mic]
-name = input-microphone-internal
-
 [Option Capture Source:Int DMic]
 name = input-microphone-internal
 
-[Option Capture Source:Internal Mic]
-name = input-microphone-internal
-
 [Option Capture Source:iMic]
 name = input-microphone-internal
 
@@ -159,15 +126,9 @@ name = input-microphone-internal
 [Option Capture Source:Internal Microphone]
 name = input-microphone-internal
 
-[Option Capture Source:Front Mic]
-name = input-microphone
-
 [Option Capture Source:Front Microphone]
 name = input-microphone
 
-[Option Capture Source:Rear Mic]
-name = input-microphone
-
 [Option Capture Source:Mic1]
 name = input-microphone
 
@@ -198,9 +159,6 @@ name = input-linein
 [Option Capture Source:Analog]
 name = input
 
-[Option Capture Source:Line]
-name = input-linein
-
 [Option Capture Source:Line-In]
 name = input-linein
 
@@ -261,9 +219,6 @@ name = input
 [Option Capture Source:Docking-Station]
 name = input-docking
 
-[Option Capture Source:Dock Mic]
-name = input-docking-microphone
-
 ;;; 'Mic Jack Mode'
 
 [Element Mic Jack Mode]
-- 
1.7.1

